python/lsst/ctrl/pool/parallel.py

```diff
@@ -473,3 +477,68 @@ def parseAndRun(cls, *args, **kwargs):
         super(BatchPoolTask, cls).parseAndRun(*args, **kwargs)
         pool.exit()
 
+
+class BatchTaskRunner(TaskRunner):
+    """Run a Task individually on a list of inputs using the MPI process pool"""
+    def __init__(self, *args, **kwargs):
+        TaskRunner.__init__(self, *args, **kwargs)
+        if self.numProcesses > 1:
+            self.log.warn("Multiprocessing arguments (-j %d) ignored since using batch processing" %
+                          self.numProcesses)
+            self.numProcesses = 1
+
+    def run(self, parsedCmd):
+        """Run the task on all targets"""
+        resultList = None
```

Comment:
This can be either `None` or `[]` on return when no data is run, depending on what happens later.  Shouldn't it be consistent, or do the two have different semantic meanings?